### PR TITLE
Polygon enable cUSDCv3 proposal

### PR DESCRIPTION
**Pull request by scott-silver**
- [x] add description
- [x] add actual supply cap values
- [x] update reward speed
- [x] update ENS text record

### CODE REVIEW

**Review by jflatow — COMMENTED** [review]
Looking good 💯

File: tasks/deployment_manager/task.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ async function runMigration<T>(
 
   if (enact) {
     console.log('Running enactment step with artifact...', artifact);
-    await migration.actions.enact(governanceDeploymentManager, artifact);
+    await migration.actions.enact(deploymentManager, governanceDeploymentManager, artifact);
```
Comment (scott-silver) [inline tasks/deployment_manager/task.ts:55]: most important part of this change: `Migration#enact` is now called with both the local deployment manager and the governance deployment manager

**Review by jflatow — APPROVED** [review]
Awesome, this is running ✅  for me

**Comment by scott-silver** [comment]
Currently, this proposal just bumps the supply caps from zero.

Is it necessary to take any actions on the configurator for the market to go live?

Do we need to call `configurator.setFactory` or `configurator.setConfiguration`, as in the WETH initialization proposal?

https://github.com/compound-finance/comet/pull/587/files#diff-c539c9bc1d4bfdfbc885fbf813eaf47173bb9ea4c4276257802c60008eaf9a08R44

File: deployments/polygon/usdc/migrations/1675200105_raise_polygon_supply_caps.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      [comet.address, WETH.address, exp(10_000, 18)] // XXX add actual amount
+    );
+    const wmaticCalldata = utils.defaultAbiCoder.encode(
+      ['address', 'address', 'uint128'],
+      [comet.address, WMATIC.address, exp(10_000, 18)] // XXX add actual amount
+    );
+    const deployAndUpgradeToCalldata = utils.defaultAbiCoder.encode(
+      ['address', 'address'],
+      [configurator.address, comet.address]
+    );
```
Comment (jflatow) [inline deployments/polygon/usdc/migrations/1675200105_raise_polygon_supply_caps.ts:46]: I think we should stick to the inline list with comments for each action format on these on all the other proposal, this format requires a lot of jumping around to read

**Comment by jflatow** [comment]
> Currently, this proposal just bumps the supply caps from zero.
> 
> Is it necessary to take any actions on the configurator for the market to go live?
> 
> Do we need to call `configurator.setFactory` or `configurator.setConfiguration`, as in the WETH initialization proposal?
> 
> https://github.com/compound-finance/comet/pull/587/files#diff-c539c9bc1d4bfdfbc885fbf813eaf47173bb9ea4c4276257802c60008eaf9a08R44

Yeah I think we should need to set the factory? Funny that its passing scenarios I think... looking to why 👀

**Comment by jflatow** [comment]
> > Currently, this proposal just bumps the supply caps from zero.
> > Is it necessary to take any actions on the configurator for the market to go live?
> > Do we need to call `configurator.setFactory` or `configurator.setConfiguration`, as in the WETH initialization proposal?
> > https://github.com/compound-finance/comet/pull/587/files#diff-c539c9bc1d4bfdfbc885fbf813eaf47173bb9ea4c4276257802c60008eaf9a08R44
> 
> Yeah I think we should need to set the factory? Funny that its passing scenarios I think... looking to why 👀

Oh yeah, we do that on deploy automatically if we can, we had to do it for WETH bc we already handed over control of Configurator to gov

**Review by kevincheng96 — APPROVED** [review]
Terrific, looks great!

File: deployments/polygon/usdc/configuration.json [hunk 3]
```diff
@@ -5,6 +5,7 @@
   "baseTokenAddress": "0x2791bca1f2de4661ed88a30c99a7a9449aa84174",
   "baseTokenPriceFeed": "0xfE4A8cc5b5B2366C1B58Bea3858e81843581b2F7",
   "borrowMin": "100e6",
+  "pauseGuardian": "0x8Ab717CAC3CbC4934E63825B88442F5810aAF6e5",
```
Comment (jflatow) [inline deployments/polygon/usdc/configuration.json:8]: Note: Waiting for multi-sig to finish rotating out the Gauntlet SAFE

File: deployments/polygon/usdc/migrations/1675200105_configurate_and_ens.ts [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+        [setConfigurationCalldata, deployAndUpgradeToCalldata]
+      ]
+    );
+
+    const RootChainManager = await deploymentManager.existing(
+      'RootChainManager',
+      RootChainManagerAddress
+    );
+    const USDCAmountToBridge = exp(400_000, 6); // roughly half of what's currently in the Timelock
+    const COMPAmountToBridge = exp(25_000, 18);
```
Comment (kevincheng96) [inline deployments/polygon/usdc/migrations/1675200105_configurate_and_ens.ts:102]: I thought we had updated these amounts. Did the rebase remove those changes?

Comment (kevincheng96) [reply deployments/polygon/usdc/migrations/1675200105_configurate_and_ens.ts:102]: Opened a PR here: https://github.com/compound-finance/comet/pull/697

I think we just never updated this after agreeing on the amounts offline 😅

Comment (kevincheng96) [reply deployments/polygon/usdc/migrations/1675200105_configurate_and_ens.ts]: Merged into this branch.

File: scenario/LiquidationBotScenario.ts [hunk 5]
```diff
@@ -604,7 +627,7 @@ scenario(
     upgrade: {
       targetReserves: exp(20_000, 18)
     },
-    filter: async (ctx) => matchesDeployment(ctx, [{deployment: 'mainnet'}]), // XXX enable for Polygon
+    filter: async (ctx) => matchesDeployment(ctx, [{network: 'mainnet'}]), // XXX enable for Polygon
```
Comment (jflatow) [inline scenario/LiquidationBotScenario.ts:630]: When should we enable this?

**Comment by kevincheng96** [comment]
Proposal enacted in this CI run: https://github.com/compound-finance/comet/actions/runs/4286566303

File: scenario/LiquidationBotScenario.ts (hunk 5 above)
Comment (kevincheng96) [reply scenario/LiquidationBotScenario.ts:630]: Good call out. Don't see a reason not to. WDYT🥈?

Comment (scott-silver) [reply scenario/LiquidationBotScenario.ts:630]: There's nothing, standing in our way to enable this for Polygon. I just suspect that it's going to be even more flakey than the other Liquidation Bot scenarios (which are already pretty flakey).

**Pull request merged**